dandiapi/api/views/webdav/__init__.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from django.shortcuts import get_object_or_404
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework.decorators import api_view
+from rest_framework.exceptions import PermissionDenied
+
+from dandiapi.api.asset_paths import get_path_children
+from dandiapi.api.models.asset_paths import AssetPath
+from dandiapi.api.models.version import Version
+from dandiapi.api.services.permissions.dandiset import is_dandiset_owner
+from dandiapi.api.views.common import PAGINATION_PARAMS
+from dandiapi.api.views.pagination import DandiPagination
+
+from .serializers import AtPathQuerySerializer, PathResultSerializer
+
+if TYPE_CHECKING:
+    from django.db.models import QuerySet
+    from rest_framework.request import Request
+
+    from dandiapi.api.models.dandiset import Dandiset
+
+
+def get_atpath_queryset(*, version: Version, path: str, children: bool) -> QuerySet[AssetPath]:
+    select_related_clauses = ('asset', 'asset__blob')
+    qs = (
+        AssetPath.objects.select_related(*select_related_clauses)
+        .filter(version=version)
+        .order_by('path')
+    )
+
+    # Handle root path case explicitly
+    if path == '':
+        return qs.exclude(path__contains='/') if children else qs.none()
+
+    # Perform path filter
+    qs = qs.filter(path=path)
+
+    # Ensure queryset isn't empty
+    asset_path = qs.first()
+    if asset_path is None:
+        return qs.none()
+
+    # Since path+version combinations are unique, we know we've matched exactly one path.
+    # Now see if we should extend this with it's children
+    if asset_path.asset is not None or not children:
+        return qs
+
+    # Now we know path is a folder, and we should show its children
+    children_paths = (
+        get_path_children(asset_path).select_related(None).select_related(*select_related_clauses)
```

Comment:
Is the intent of the first `select_related` to "remove" the `select_related` that's added by `get_path_children`? And if so, have you tested that it works? I've never seen this pattern before, but it makes sense to me.

Reply:
Yes, that's the intention, and it does work as expected. It is mentioned [here](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#:~:text=without_relations%20%3D%20queryset.select_related(None)) in the django docs.